app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/PoolContributionProcessor.kt

```diff
@@ -21,37 +21,40 @@ import rdx.works.profile.domain.accountsOnCurrentNetwork
 import javax.inject.Inject
 
 class PoolContributionProcessor @Inject constructor(
-    private val getResourcesUseCase: GetResourcesUseCase,
+    private val resolveAssetsFromAddressUseCase: ResolveAssetsFromAddressUseCase,
     private val getPoolDetailsUseCase: GetPoolDetailsUseCase,
     private val getProfileUseCase: GetProfileUseCase
 ) : PreviewTypeProcessor<DetailedManifestClass.PoolContribution> {
     override suspend fun process(summary: ExecutionSummary, classification: DetailedManifestClass.PoolContribution): PreviewType {
-        val resources = getResourcesUseCase(addresses = summary.involvedResourceAddresses).getOrThrow()
+        val assets = resolveAssetsFromAddressUseCase(
+            fungibleAddresses = summary.involvedFungibleAddresses,
+            nonFungibleIds = summary.involvedNonFungibleIds
+        ).getOrThrow()
         val involvedPools = getPoolDetailsUseCase(classification.poolAddresses.map { it.addressString() }.toSet()).getOrThrow()
-        val defaultDepositGuarantees = getProfileUseCase.invoke().first().appPreferences.transaction.defaultDepositGuarantee
+        val defaultDepositGuarantee = getProfileUseCase.invoke().first().appPreferences.transaction.defaultDepositGuarantee
         val accountsWithdrawnFrom = summary.accountWithdraws.keys
         val ownedAccountsWithdrawnFrom = getProfileUseCase.accountsOnCurrentNetwork().filter {
             accountsWithdrawnFrom.contains(it.address)
         }
-        val from = summary.extractWithdraws(ownedAccountsWithdrawnFrom, resources)
+        val from = summary.extractWithdraws(ownedAccountsWithdrawnFrom, assets.map { it.resource })
         val to = summary.accountDeposits.map { depositsPerAddress ->
             val ownedAccount = getProfileUseCase.accountOnCurrentNetwork(depositsPerAddress.key) ?: error("No account found")
-            val deposits = depositsPerAddress.value.mapNotNull { deposit ->
+            val deposits = depositsPerAddress.value.map { deposit ->
                 val resourceAddress = deposit.resourceAddress
                 val contributions = classification.poolContributions.filter {
                     it.poolUnitsResourceAddress.addressString() == resourceAddress
                 }
                 if (contributions.isEmpty()) {
-                    null
+                    resolveGeneralAsset(deposit, summary, assets, defaultDepositGuarantee)
```

Comment:
Why is this needed here? Can you explain when we have empty contributions?

Reply:
see second bullet

Reply:
if you contribute to pool amounts of resources that do not convert to pool units entirely (for example you have ratio of resources A:B as 1:2, and you contribute 1 unit of A and 2.5 of B (via sandbox), that reminder of 0.5 B is returned by RET as regular resource deposited to the specific account. Without that change, this is not reflected in the "Depositing" section, although you see in "Withdrawn" 2.5 of B being withdrawn.

Reply:
Ah I see now thanks